test/blackbox/BlackboxTestsTransportSHM.cpp

```diff
@@ -0,0 +1,349 @@
+// Copyright 2020 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "BlackboxTests.hpp"
+
+#include "PubSubReader.hpp"
+#include "PubSubWriter.hpp"
+
+#include <gtest/gtest.h>
+
+#include <fastrtps/transport/UDPv4Transport.h>
+#include "../cpp/rtps/transport/shared_mem/test_SharedMemTransportDescriptor.h"
+
+using namespace eprosima::fastrtps;
+
+using SharedMemTransportDescriptor = eprosima::fastdds::rtps::SharedMemTransportDescriptor;
+using test_SharedMemTransportDescriptor = eprosima::fastdds::rtps::test_SharedMemTransportDescriptor;
+
+TEST(BlackBox, SHMTransportPubSub)
```

Comment:
Maybe add all tests to a 'SharedMemory' block, instead of 'BlackBox'

Applies to all tests on this file

Reply:
OK. But I will put SHM instead SharedMemory because all shared-memory test include SHM in the name, that way is very easy to filter with ctest.